### PR TITLE
Upgrades: fix Eslint warnings. ES6ify.

### PR DESCRIPTION
**Pull request by retrofox**
This PRs just fixes many Eslint warnings and implement ES6 syntax.

### Testing

The changes affect to the payment boxes. One way to test these ones is making the flow to upgrades a plan in the different ways that we already have.

image

Test live: https://calypso.live/?branch=update/janitorial-my-sites-checkout

### CODE REVIEW

File: client/my-sites/upgrades/checkout/credit-card-payment-box.jsx [hunk 1]
```diff
@@ -1,62 +1,67 @@
...
+import PayButton from './pay-button';
+import CreditCardSelector from './credit-card-selector';
+import TermsOfService from './terms-of-service';
+import PaymentBox from './payment-box';
+import analytics from 'lib/analytics';
+import cartValues from 'lib/cart-values';
 
-var CreditCardPaymentBox = React.createClass( {
-	getInitialState: function() {
+const CreditCardPaymentBox = React.createClass( {
```
Comment (lamosty) [inline client/my-sites/upgrades/checkout/credit-card-payment-box.jsx]: Why not using the ES6 class syntax here?

```js

class CreditCardPaymentBox extends Component { ... }
```

File: client/my-sites/upgrades/checkout/credit-card-payment-box.jsx [hunk 2]
```diff
@@ -1,62 +1,67 @@
...
 						cart={ this.props.cart }
 						transactionStep={ this.props.transactionStep } />
 
-					{ cartValues.isPayPalExpressEnabled( cart )
-						? <a className="credit-card-payment-box__switch-link" href="" onClick={ this.handleToggle }>{ this.translate( 'or use PayPal' ) }</a>
-						: null
+					{ cartValues.isPayPalExpressEnabled( cart ) &&
+						<a
+							className="checkout__credit-card-payment-box__switch-link"
+							href=""
```
Comment (lamosty) [inline client/my-sites/upgrades/checkout/credit-card-payment-box.jsx]: Does it need an empty `href` prop?

**Comment by lamosty** [comment]
Not sure if it's related to changes in this PR, but I'm seeing something strange on the credit card secure payment screen:

selection_051

File: client/my-sites/upgrades/checkout/credit-card-payment-box.jsx (hunk 2 above)
Comment (retrofox) [inline client/my-sites/upgrades/checkout/credit-card-payment-box.jsx]: I've just ordered the line here. I'm not sure what's the reason of these empty property. The PR has a janitorial purpose.

**Comment by retrofox** [comment]
I'll take a look of that. Nice catch.

File: client/my-sites/upgrades/checkout/credit-card-payment-box.jsx [hunk 3]
```diff
@@ -1,62 +1,67 @@
...
+							onClick={ this.handleToggle }
+						>
+							{ this.translate( 'or use PayPal' ) }
```
Comment (lamosty) [inline client/my-sites/upgrades/checkout/credit-card-payment-box.jsx]: Now that we have `i18n-calypso`, I think it's better to write this code like:

```js
import { localize } from 'i18n-calypso';

class XY extends Component { 

render() {
   const { translate } = this.props;

  return ( translate( 'some string' );
}
}

export default localize( XY );
```

File: client/my-sites/upgrades/checkout/credit-card-payment-box.jsx (hunk 2 above)
Comment (lamosty) [inline client/my-sites/upgrades/checkout/credit-card-payment-box.jsx]: >  The PR has a janitorial purpose.

I see, but maybe we could get rid of unneeded things when we are tidying it up. :)

File: client/my-sites/upgrades/checkout/free-cart-payment-box.jsx [hunk 4]
```diff
@@ -1,69 +1,77 @@
...
-						cart.has_bundle_credit ?
-							this.translate( 'You have a free domain credit!' ) :
-							this.translate( "Woohoo! You don't owe us anything!" ) }
+			<form onSubmit={ onSubmit }>
+				<div className="checkout__payment-box-section">
+					<h6>
+						{
+							cart.has_bundle_credit
+							? this.translate( 'You have a free domain credit!' )
+							: this.translate( "Woohoo! You don't owe us anything!" )
```
Comment (lamosty) [inline client/my-sites/upgrades/checkout/free-cart-payment-box.jsx]: I would give this one tab so it's clear that it's a continued expression.

```js

cart.has_bundle_credit
    ? this....
    : this...
```

File: client/my-sites/upgrades/checkout/credit-card-payment-box.jsx (hunk 2 above)
Comment (retrofox) [inline client/my-sites/upgrades/checkout/credit-card-payment-box.jsx]: Yes, but I've experimented this situation before. I'm trying to keep the scope of this PR. The idea is remove the all (annoying but useful) ESLint warnings and implement ES6 syntax.

We could take a look of this question tough. :-D

Comment (lamosty) [inline client/my-sites/upgrades/checkout/credit-card-payment-box.jsx]: Alright, then.:)

**Comment by gwwar** [comment]
This one still active @retrofox?

**Comment by retrofox** [comment]
Close this one in favor to #7507.

**Pull request closed without merging**